nixos/tests/initrd-secrets-changing.nix

```diff
@@ -0,0 +1,55 @@
+{ system ? builtins.currentSystem
+, config ? {}
+, pkgs ? import ../.. { inherit system config; }
+, lib ? pkgs.lib
+, testing ? import ../lib/testing-python.nix { inherit system pkgs; }
+}:
+
+let
+  secret1InStore = pkgs.writeText "topsecret" "iamasecret1";
+  secret2InStore = pkgs.writeText "topsecret" "iamasecret2";
+in
+
+testing.makeTest {
+  name = "initrd-secrets-changing";
+
+  nodes.machine = { ... }: {
+    virtualisation.useBootLoader = true;
+    boot.loader.grub.device = "/dev/vda";
+
+    boot.initrd.secrets = {
+      "/test" = secret1InStore;
+      "/run/keys/test" = secret1InStore;
+    };
+    boot.initrd.postMountCommands = "cp /test /mnt-root/secret-from-initramfs";
+
+    specialisation.secrets2System.configuration = {
+      boot.initrd.secrets = lib.mkForce {
+        "/test" = secret2InStore;
+        "/run/keys/test" = secret2InStore;
+      };
+    };
+  };
+
+  testScript = ''
+    start_all()
+
+    machine.wait_for_unit("multi-user.target")
+    print(machine.succeed("cat /run/keys/test"))
+    machine.succeed(
+        "cmp ${secret1InStore} /secret-from-initramfs",
+        "cmp ${secret1InStore} /run/keys/test",
+    )
+    machine.succeed("grub-reboot 1")
```

Comment:
I had to read the test twice to spot that this reboots into the specialisation 2 (second entry in Grub, numbered from zero). 

According to the `grub-reboot` man page this could also be a menu item title - more readable, but probably unpredictable?

I think at least a comment would be nice, though!

Reply:
For this I copied the pattern from the nixos-installer tests:

https://github.com/NixOS/nixpkgs/blob/af0b453724a59bc8fa90fa1dfad30896cc3df529/nixos/tests/installer.nix#L243-L244

You're right that I predicting the correct boot string is harder. Doable, but I figured from looking at the `grub.cfg` and knowing how the `grub.cfg` is constructed that taking the second entry would be the more straightforward and reliable approach.